frontend/src/modules/api/core.ts

```diff
@@ -0,0 +1,101 @@
+import { MIME_SUBTYPE, MIME_TYPE, RESPONSE_TYPE } from "./constant";
+
+interface IApiOptions {
+  method?: string;
+  headers?: {
+    "Content-Type"?: string;
+    Authorization?: string;
+  };
+  body?: string;
+}
+
+interface IApiResponse<T> {
+  statusCode: number;
+  data: T;
+}
+
+export const api = <T>(
+  url: string,
+  options?: IApiOptions
+): Promise<IApiResponse<T>> => {
+  const xhr = new XMLHttpRequest();
+  return new Promise((resolve, reject) => {
+    setXhrOptions(xhr, url, options);
+    xhr.onreadystatechange = function () {
+      if (xhr.readyState === XMLHttpRequest.DONE) {
+        const status = xhr.status;
+        const { type } = parseContentType(
+          xhr.getResponseHeader("Content-type")
+        );
+
+        isOk(status)
+          ? resolve(parseResponse<T>(type, xhr))
+          : reject({ statusCode: xhr.status, data: xhr.response });
+      }
+    };
+
+    if (options) {
+      if (options.method === "GET" || options.method === "DELETE") {
+        xhr.send();
+      } else {
+        xhr.send(options.body || "");
+      }
+    } else {
+      xhr.send();
+    }
```

Comment:
```suggestion
    xhr.send(options?.body || null);
```

이런 식으로 줄일 수도 있지 않을까요?